oteapi_dlite/strategies/parse_image.py

```diff
@@ -0,0 +1,195 @@
+"""Strategy class for parsing xlsx to a DLite instance."""
+# pylint: disable=no-self-use,unused-argument
+from dataclasses import dataclass
+from pathlib import Path
+from random import getrandbits
+
+import numpy as np
+from PIL import Image
+from pydantic import BaseModel, Field, HttpUrl
+from typing import TYPE_CHECKING, Any, Dict, Optional, Tuple
+
+import dlite
+
+if TYPE_CHECKING:  # pragma: no cover
+    from typing import Any, Dict, Optional, Tuple
+
+    from oteapi.models.resourceconfig import ResourceConfig
+
+
+class DLiteImageConfig(BaseModel):
+    """Configuration for DLite image parser."""
+
+    def __init__(self, **kwargs) -> None:
+        """Initialize image configuration object."""
+        super().__init__()
+        if not kwargs:
+            return
+        config = kwargs.copy()
+        if "crop" in config:
+            self.crop = config.pop("crop", self.crop)
+        if "id" in config:
+            self.id = config.pop("id", self.id)
+        if "metadata" in config:
+            self.metadata = config.pop("metadata", self.metadata)
+        if config:
+            self.configuration = config
+
+    configuration: Optional[Dict[str, Any]] = Field(
+        None, description="Specific image configuration parameters."
+    )
+
+    crop: Optional[Tuple] = Field(
+        None, description="Cropping rectangle. The whole image if None."
+    )
+
+    id: Optional[str] = Field(None, description="Optional id for new instance.")
+
+    metadata: Optional[HttpUrl] = Field(
+        None,
+        description=(
+            "URI of DLite metadata to return.  If not provided, the metadata "
+            "will be inferred from the image file."
+        ),
+    )
+
+
+@dataclass
+class DLiteImageParseStrategy:
+    """Parse strategy for image files.
+
+    **Registers strategies**:
+
+    - `("mediaType", "image/gif")`
+    - `("mediaType", "image/jpeg")`
+    - `("mediaType", "image/jpg")`
+    - `("mediaType", "image/jp2")`
+    - `("mediaType", "image/png")`
+    - `("mediaType", "image/tiff")`
+
+    """
+
+    META_PREFIX = "http://onto-ns.com/meta/1.0/generated_from_"
+    parse_config: "ResourceConfig"
+
+    def initialize(
+        self, session: "Optional[Dict[str, Any]]" = None
+    ) -> "Dict[str, Any]":
+        """Initialize."""
+        return {}
```

Comment:
Nowdays (that is since Friday) all filters should return a SessionUpdate()
```suggestion
        return SessionUpdate()
```